misk-aws/src/main/kotlin/misk/jobqueue/sqs/MiskJobModule.kt

```diff
@@ -0,0 +1,124 @@
+package misk.jobqueue.sqs
+
+import com.google.common.util.concurrent.Service
+import misk.environment.Environment
+import misk.inject.KAbstractModule
+import misk.inject.asSingleton
+import misk.inject.toKey
+import misk.jobqueue.JobConsumer
+import misk.jobqueue.JobHandler
+import misk.jobqueue.JobQueue
+import misk.jobqueue.QueueName
+import java.time.Duration
+import javax.inject.Provider
+import kotlin.reflect.KClass
+
+/**
+ * Install multiple job queues backed by AwsSQS.
+ *
+ * Usage
+ *
+ * In your misk service Module include:
+ *  ```
+ *  install(MiskJobModule(environment)
+ *    .withHandler<SomeJobHandlerA>(JobQueueA::class, "job_queue_a")
+ *    .withHandler<SomeJobHandlerB>(JobQueueB::class, "job_queue_b"))
+ *
+ *  ```
+ *  and later access using:
+ *  ```
+ *  @Inject @JobQueueA lateinit var jobQueueA: MiskJobQueue`
+ *  ```
+ */
+
+class MiskJobModule(private val environment: Environment) : KAbstractModule() {
+  private lateinit var jobQueueProvider: Provider<JobQueue>
+  private val jobQueuesMap =
+      mutableListOf<JobQueueBinding>()
+
+  inline fun <reified T : JobHandler> withHandler(
```

Comment:
can we make the annotation be a `reified` type parameter as well?

Reply:
yes! didn't know this was possible lol